### ISSUE TITLE
v5 RestApiClientAsync much slower than v4 RestApiClient and cause connect_timeout

### ISSUE THREAD

**Issue by subbyte** [opencybersecurityalliance/stix-shifter#1493]
**Describe the bug**
The `aiohttp` or `aiohttp-retry` library in stix-shifter v5 looks very slow and cause `connect_timeout`, which is not an issue in stix-shifter v4.

**To Reproduce**
Run the STIX query `[ipv4-addr:value = '127.0.0.1']` or the following equivalent Kestrel huntflow:
```
ips = GET ipv4-addr FROM stixshifter://abc
      WHERE value = '127.0.0.1'
      START 2022-10-01T00:00:00Z STOP 2023-05-01T00:00:00Z
```
We have more than 55k data in an Elasticsearch store for this query to retrieve.

[note] STIX query triggered by the stix-shifter CLI
- does not have pagination
- does not retrieve all results
- so it does not reveal the issue described below in a real app

The Kestrel huntflow with the following setting
```
            options:
                result_limit: 500000
                retrieval_batch_size: 10000
                timeout: 60
```
will be able to retrieve all data in 5 batches. The rest api `call()` in [stix_shifter_utils/stix_transmission/utils/RestApiClientAsync.py](https://github.com/opencybersecurityalliance/stix-shifter/blob/develop/stix_shifter_utils/stix_transmission/utils/RestApiClientAsync.py#L122) will be invoked 6 times---1 for page length probing, 5 for `transmit.results()`.

A few observations:

1. When print out each time the `call()` function is invoked and finished, I see significant more time for it to run in v5 compared with v4. This results in twice amount of time to execute the Kestrel huntflow with stix-shifter v5 than v4.
2. When the 6 `call()` API are executed in a sequence, each one takes more time than the previous `call()`. The time consumed plateaus at some point, e.g., after 4 `call()`.
3. At some point during the 6 `call()` API execution, we could see an error message
```
STIX-shifter transmission.results() failed with message: elastic_ecs connector error => server timeout_error (2 sec)
```
4. The time to execute a `call()` function changes when I change `retrieval_batch_size`---it tells Elasticsearch how many documents/records to return---This cause the error to happen or not to happen (since it is thresholding on time). Smaller `retrieval_batch_size` takes less time to execute the `call()` function and have less chance to emit the `connect_timeout` error. I thought the `connect_timeout` error is not for entire `timeout`, but just for connection establishment, and this parameter in the HTTP request should not affect anything there. So I am confused.

This error is due to the very small `CONNECT_TIMEOUT_DEFAULT = 2` set in [RestApiClientAsync.py](https://github.com/opencybersecurityalliance/stix-shifter/blob/develop/stix_shifter_utils/stix_transmission/utils/RestApiClientAsync.py#L18). However, increasing the number helps but not solve the unknown underlying problem. It just gets less chance to occur---the error will occur later in the 6 `call()` sequence, e.g., in the third, not second `call()`. If it is large enough, e.g., removing the `connect_timeout`, it is executed fine, but the execution time is large as I mentioned in the first bullet.

It looks to me there is something incorrect underlying the async rest API call. That is why v4 takes less time to run than v5 and never hit the `connect_timeout` error.

**Expected behavior**
v5 to run as fast as v4, and does not hit the `connect_timeout` error.

**Environment (please complete the following information):**
 - OS: macOS and Linux
 - stix-shifter: 5.3.0 and 4.6.5 tested and compared in the description
 - kestrel-lang: 1.6.0 and 1.5.14 tested with the corresponding stix-shifter v5/v4

**Comment by cmadam** [comment]
The most likely reason behind this issue is that, when we are running the two steps of the pipeline (transmission and translation) in parallel , if the CPU is overloaded with the translation task, the timed transmission operation to connect to the server takes longer than 2 seconds, and consequently it throws the timeout error.

We can simulate an overloaded CPU by placing a `time.sleep(1.99)`  instruction at line 1139 of the file `lib/python3.8/site-packages/aiohttp/connector.py`, in the `_create_direct_connection` method of the `TCPConnector` class.  That `sleep` statement will always make it fail with the error described in the bug report.

**Comment by subbyte** [comment]
The root cause of the issue is confirmed:
1. stix-shifter transmission is I/O-bound, but stix-shifter translation is CPU-bound
2. since stix-shifter translation is CPU-bound, calling it through asyncio does not release the CPU cycle from it
3. the continues CPU usage of translation will timeout the transmission that is initialized and waiting

To avoid the issue, the upper-layer application that calls stix-shifter needs to make the I/O-bound transmission and CPU-bound translation working together, which is possible either:
1. using multi-threading: this is possible with `asyncio`, basically putting I/O-bound tasks as regular asyncio function and CPU-bound tasks into `run_in_executor()`. Here's a [reference implementation in Kestrel](https://github.com/opencybersecurityalliance/kestrel-lang/pull/351/files )(abandoned in favor of the multi-processing approach).
2. using multi-processing: this is much cleaner and has better performance with multiple CPU-bound tasks. kestrel-lang>=1.7.0 moves to this implementation, explained in the [Kestrel data retrieval blog](https://opencybersecurityalliance.org/kestrel-data-retrieval-explained/).

**Comment by subbyte** [comment]
solved in Kestrel v1.7.0
